packages/modules/payment/src/services/payment-module.ts

```diff
@@ -965,17 +973,20 @@ export default class PaymentModuleService
   ): Promise<PaymentMethodDTO | PaymentMethodDTO[]> {
     const input = Array.isArray(data) ? data : [data]
 
-    const result = await Promise.all(
+    const result = await promiseAll(
       input.map((item) =>
         this.paymentProviderService_.savePaymentMethod(item.provider_id, item)
-      )
+      ),
+      { aggregateErrors: true }
     )
 
-    const normalizedResponse = result.map((item, i) => ({
-      id: item.id,
-      data: item.data,
-      provider_id: input[i].provider_id,
-    }))
+    const normalizedResponse = result
+      .map((item, i) => ({
+        id: item?.id,
+        data: item?.data,
+        provider_id: input[i].provider_id,
+      }))
+      .filter((item) => !!item.id && !!item.data) as PaymentMethodDTO[]
```

Comment:
Q: In whish case we end up with no item id and item data?

Reply:
This is a bit messy I guess... the return of `this.paymentProviderService_.savePaymentMethod(item.provider_id, item)` can either be an object or undefined. 

Now in the normalization, I leave `id` and `data` as undefined if the response for that entry was undefined, because I then use the index of the array to get the provider ID, and only after I filter the results that had undefined response from the provider.

Let me see how I can refactor it.

Reply:
My thinking is that we are kind of hiding that a provider does not support payment methods no? Would it be possible that we add a method in the provider service to check if payment methods support is available and otherwise throw an error to say that X and Y are trying to create payment methods on provider Z while it is not supported?

Reply:
Pushed a change, LMK what you think and if you have a suggestion how to handle it better. The only annoying thing is that you might pass eg. 5 entries on input but get 2 back, but I don't know how better to handle it TBH

Reply:
I am worried the interface will become a bit bloated with that sort of check. Throwing though might make sense here, and then the caller can catch that and handle it as they deem fit.

Reply:
Because here we would be in the scenario where you try to create a payment methods but it is not supported, so instead I would throw from the provider (wdyt) instead of returning undefined to indicate that the user is trying to create a payment method on a provider that does not support it

Reply:
yes exactly, and since we aggregate the errors, all supported providers will create the payment methods and you will get the error for the one that does not support it or something went wrong

Reply:
Yeah let's do that

Reply:
> interface will become a bit bloated with that sort of check.

It wouldn't be on the provider, but on the module provider service, but as per my second suggestion it would not be necessary

Reply:
Pushed it, have a look

Reply:
perfect 👍

Reply:
should we add a few tests?